### ISSUE TITLE
layer heights problem

### ISSUE THREAD

**Issue by legend069** [supermerill/SuperSlicer#1992]
### Version
Version 2.3.57

### Operating system
Operating System: Windows
System Architecture: 64 bit
Windows Version: Windows 10 (build 19042), 64-bit edition

### Behavior
* problem: 
  * layer height is adjusted by bridge flow ratio
* Steps needed to reproduce the problem: 
  * adjusting bridge flow ratio then looking at the layer height
* _Expected Results:
  *  it to not change 
* _Actual Results_ layer height changes:
  *  messes with other parts of print if using adaptive layers

_Is this a new feature request?_ no

#### Project File (.3MF) where problem occurs
_*Upload a Project File* (.3MF) `Plater -> Save Project as...`_
_You have to zip it (or rename it to .zip) to be able to drag& drop it here
![superslicer_R1ung1npmK](https://user-images.githubusercontent.com/40685552/144439612-a8af216b-79ac-4c4c-a2d5-e2300b00f7c1.png)
![superslicer_UE9bPIOyOU](https://user-images.githubusercontent.com/40685552/144439622-39542e4c-4ed8-453e-a987-544ea59c8ace.png)
![superslicer_hcEyc7LwmS](https://user-images.githubusercontent.com/40685552/144439627-12bafc61-a2be-42ad-91cf-bef806cfdb1f.png)
![superslicer_lsgAYlgIqq](https://user-images.githubusercontent.com/40685552/144439630-f7200259-5196-4f06-bad0-d4be0a3e5264.png)
[slicer_config_and_gcode_file_used.zip](https://github.com/supermerill/SuperSlicer/files/7642616/slicer_config_and_gcode_file_used.zip)

_

**Comment by supermerill** [comment]
I think you forgot to replace the template text by your own.
There is no description to read. I don't really understand what's wrong here.

**Comment by supermerill** [comment]
Please save your project file(s) : File -> save project as
It's mucgh much easier for me.

**Comment by legend069** [comment]
> Please save your project file(s) : File -> save project as
> It's much much easier for me.

Sorry I thought I did that.
I'll do that in the morning

**Comment by legend069** [comment]
I've added the file into the zipped folder

[slicer_config_and_gcode_file_used.zip](https://github.com/supermerill/SuperSlicer/files/7647556/slicer_config_and_gcode_file_used.zip)

**Comment by supermerill** [comment]
Can't spot anything wrong
Changing the flow ratio of the bridge modify the height & width of it, and nothing else.

**Comment by legend069** [comment]
> o of the bridge

so if i have " flow ratio of the bridge" set to 80%
and adaptive layers enabled  this would be changing z height to compensate for the bridge area that is 0.360 and then back down to 0.100 correct?

then for the travel movements across that area  z height change ?

hopefully that makes sense


![superslicer_jgkgbOhl3f](https://user-images.githubusercontent.com/40685552/144590208-3f5f0357-73f3-415d-a6fa-e32f399f94c7.png)

**Comment by supermerill** [comment]
sorry i don't understand.

**Comment by legend069** [comment]
> sorry i don't understand.

in attached screenshots, the nozzle moves overtop of the area that is a different layer height but the z position doesn't go up to go over the height layer



![superslicer_G4HYzlhfbt](https://user-images.githubusercontent.com/40685552/144590824-92a67621-8ca2-4747-95cd-14493587060f.png)
![superslicer_mPs6wki9ou](https://user-images.githubusercontent.com/40685552/144590833-e2b3a9dc-b23c-49bf-ba01-98200c2e325c.png)

**Comment by neophyl** [comment]
Merill I think they are having issue with the fact that for bridging the layer height is equal to nozzle width and not 0.1 like the rest of the layers.
What they (and most people it seems) don't understand that ANYTHING printed as a bridge is not squished against the layer below and so its modeled as a height=width as that's how the plastic comes out the nozzle.  I think they were expecting the bridge layers to be printed higher up as with a thicker height to make the bottom match the model they would need to be.  

I think the constant stream of issues relating to this function was what finally made Prusa have the option to change this behavior in their 2.4 builds.  I must admit in all the printing I've done I always turned that off to get rid of the bridge flow wherever possible as it always printed better without.  

Basically most of these issue will go away after the 2.4 merge and all we will be doing after that is constantly pointing out which setting to change to use the thinner bridge height.

**Comment by supermerill** [comment]
I was also thinking about moving the bridge layer a bit higher to be sure it doesn't overflow on the empty area... later.

@legend069 sadely, no, that's how it works. You can reduce the bridge flow to reduce the overflow.

**Comment by celtare21** [comment]
> I was also thinking about moving the bridge layer a bit higher to be sure it doesn't overflow on the empty area... later. btw, the "bridge kill switch" is just a way to lie. It won't stop physic to create a cylinder because there is nothing under it.
> 
> @legend069 sadely, no, that's how it works. You can reduce the bridge flow to reduce the overflow.

Would asking for a port of the "thick_bridges" option from prusaslicer 2.4 be too much? Or you'd rather not mess with it as i heard there's plans to update the prusaslicer version at some point.

**Comment by supermerill** [comment]
> Would asking for a port of the "thick_bridges" option from prusaslicer 2.4 be too much? Or you'd rather not mess with it as i heard there's plans to update the prusaslicer version at some point.

There's plans to update the prusaslicer version when out of beta.

also, disbaling "thick bridge" is the same as 'please i want to ignore the physics and just want a pretty preview'.
It's the same as reducing the bridge flow anyway. I'll add some user-friendly way of tuning the flow at the same time.

**Comment by celtare21** [comment]
> > Would asking for a port of the "thick_bridges" option from prusaslicer 2.4 be too much? Or you'd rather not mess with it as i heard there's plans to update the prusaslicer version at some point.
> 
> There's plans to update the prusaslicer version when out of beta.
> 
> also, disabling "thick bridge" is the same as 'please i want to ignore the physics and just want a pretty preview'. It's the same as reducing the bridge flow anyway. I'll add some more user-friendly way of tuning the flow at the same time.

Well, for me it's not about a "pretty preview". I find that "thick bridges" are detrimental to any kind of bridge or overhang area. I understand the theoretical reasoning it is there, but at least in my opinion in practice it is quite the opposite. Bridges printed at 0.2mm look beatiful while bridges printed "thick" are way thicker than they need to be and it really shows not only making it ugly, but if this was a hole for example printed vertically, the "thick" also totally messes the dimensions and usually requires post processing.

**Comment by supermerill** [comment]
then reduce the bridge flow and decrease the bridge overlap to add some spacing. It's the same result if you enter the good values.

**Comment by celtare21** [comment]
> then reduce the bridge flow and decrease the bridge overlap to add some spacing. It's the same result if you enter the good values.

That also changes the line width, which is not wanted. I don't really see any way to replicate "thick_bridges" toggle in susi as of now.

**Comment by supermerill** [comment]
The line width is purly virtual, the gcode contains only flow.

**Comment by celtare21** [comment]
Oh, I didn't know that at all, thank you for the explication

**Comment by legend069** [comment]
thanks for clearing it up
@supermerill
